vfs/vfscache/cache.go

```diff
@@ -74,16 +74,21 @@ type AddVirtualFn func(remote string, size int64, isDir bool) error
 // This starts background goroutines which can be cancelled with the
 // context passed in.
 func New(ctx context.Context, fremote fs.Fs, opt *vfscommon.Options, avFn AddVirtualFn) (*Cache, error) {
+	fName := fremote.Name()
 	fRoot := filepath.FromSlash(fremote.Root())
 	if runtime.GOOS == "windows" {
 		if strings.HasPrefix(fRoot, `\\?`) {
 			fRoot = fRoot[3:]
 		}
 		fRoot = strings.Replace(fRoot, ":", "", -1)
+		// Remove leading ':' if remote was created on the fly as ":backend:/path"
+		if fName[0] == ':' {
+			fName = fName[1:]
```

Comment:
I think we need to replace the ':' with another character, preferably one that doesn't occur in backend names otherwise there can be collisions.

The only non-alpha characters allowed in backends is `_ -` so a non alpha character which is allowed in windows file names not from that list - how about `^` - that doesn't appear on anyone's list of illegal filename characters.